src/utils/pars_utils.jl

```diff
@@ -37,49 +37,112 @@ Write validity for a given filekey.
-    Distributed.remotecall_fetch(_writevalidity_impl, 1, props_db, filekey, apply; category=category, write_from_master= (Distributed.myid() == 1))
+function writevalidity(props_db::LegendDataManagement.MaybePropsDB, filekey::FileKey, apply::Vector{String}; category::DataCategoryLike=:all, mode::AbstractString = "reset")
+    Distributed.remotecall_fetch(_writevalidity_impl, 1, props_db, filekey, apply; category=category, mode=mode, write_from_master= (Distributed.myid() == 1))
 end
 const _writevalidity_lock = ReentrantLock()
-function _writevalidity_impl(props_db::LegendDataManagement.MaybePropsDB, filekey::FileKey, apply::Vector{String}; category::DataCategoryLike=:all, write_from_master::Bool=true)
-    # write validity
+function _writevalidity_impl(props_db::LegendDataManagement.MaybePropsDB, filekey::FileKey, apply::Vector{String};
+    category::DataCategoryLike = :all, mode::AbstractString = "reset", write_from_master::Bool = true)
+
     @lock _writevalidity_lock begin
-        # get timestamp from filekey
-        pars_validTimeStamp = string(filekey.time)
-        # get validity filename and check if exists
-        validity_filename = joinpath(data_path(props_db), validity_filename)
-        mkpath(dirname(validity_filename))
-        touch(validity_filename)
-        # check if validity already written
-        validity_lines = readlines(validity_filename)
-        # check if given validity already exists
-        is_validity = findall(x -> contains(x, "$pars_validTimeStamp") && contains(x, "$(string(category))"), validity_lines)
-        if isempty(is_validity)
-            if write_from_master @info "Write new validity for $pars_validTimeStamp" end
-            push!(validity_lines, "{\"valid_from\":\"$pars_validTimeStamp\", \"category\":\"$(string(category))\", \"apply\":[\"$(join(sort(apply), "\", \""))\"]}")
-        elseif length(is_validity) == 1
-            if write_from_master @info "Merge old $pars_validTimeStamp $(string(category)) validity entry" end
-            apply = unique(append!(Vector{String}(JSON.parse(validity_lines[first(is_validity)])["apply"]), apply))
-            validity_lines[first(is_validity)] = "{\"valid_from\":\"$pars_validTimeStamp\", \"category\":\"$(string(category))\", \"apply\":[\"$(join(sort(apply), "\", \""))\"]}"
+        # locate & create file if missing
+        dst = joinpath(data_path(props_db), "validity.yaml")
+        mkpath(dirname(dst))
+        if !isfile(dst)
+            open(dst, "w") do io
+                println(io, "[]")
+            end
         end
-        # write validity
-        open(validity_filename, "w") do io
-            for line in sort(validity_lines)
-                println(io, line)
+
+        # load or initialize the vector of entries
+        raw     = YAML.load_file(dst, dicttype = OrderedDict)
+        entries = raw isa Vector ? raw : Any[]
```

Comment:
can `raw` be anything else than `nothing`?
And I don't like `Any[]` at all... Looks like `entries` is a `OrderedDict{String,Any}[]`.

Does this work?
```suggestion
        raw     = YAML.load_file(dst, dicttype = OrderedDict{String, Any})
        entries::Vector{OrderedDict{String,Any}} = raw isa Vector ? raw : OrderedDict{String,Any}[]
```